server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -739,11 +748,13 @@ public XContentBuilder snapshotsToXContent(final XContentBuilder builder, final
                 builder.endObject();
             }
             final IndexVersion version = snapshotDetails.getVersion();
-            if (version != null) {
-                if (version.before(IndexVersion.V_8_9_0)) {
-                    builder.field(VERSION, Version.fromId(version.id()).toString());
+            if (version != null && version.id() != NUMERIC_INDEX_VERSION_MARKER.id()) {
+                if (version.onOrAfter(IndexVersion.V_8_500_000)) {
```

Comment:
Perhaps I'm missing something, but I thought the consensus reached was that this comparison should be on minVersion since that is effectively the format version of the entire repository data?

Reply:
That is not a problem with this approach, since the `NUMERIC_INDEX_VERSION_MARKER_STRING` is compatible with older releases.

Reply:
Yes, technically we're not changing the format version of the entire repository data here, we're just refining the format of the snapshot version slightly. The format of older version numbers remains unchanged, newer version numbers are formatted such that older versions will see them as 8.11.0, but newer versions will interpret them correctly.

In fact the round-tripping problem doesn't matter here, because the authoritative source of a snapshot version is the `SnapshotInfo` file, and we already have a mechanism to re-read these files if we find we do not have everything we expect.